fava/application.py

```diff
@@ -325,6 +326,17 @@ def api_add_document():
             return api_error('{} already exists.'.format(filepath))
 
         file.save(filepath)
+
+        if request.form.get('bfilename', None):
+            # TODO use next_statement_key to calculate the metadata key to
+            # support multiple statement keys
+            relpath = os.path.relpath(filepath, os.path.dirname(request.form['bfilename']))
+            insert_line_in_file(
+                request.form['bfilename'],
+                int(request.form['blineno'])-1,
+                'statement: "{}"'.format(relpath))
+
+        g.api.changed(force=True)
```

Comment:
If the change is not picked up, that's a bug either because the watcher doesn't work correctly or that maybe the reloading is called too early from Javascript. Let's not hide bugs with hacks.